modules/ipi-install-troubleshooting-troubleshooting-failure-to-initialize-the-cluster.adoc

```diff
@@ -0,0 +1,191 @@
+
+.Procedure
+
+. Inspect the `ClusterVersion` object by running the following command:
++
+[source,terminal]
+----
+$ oc --kubeconfig=${INSTALL_DIR}/auth/kubeconfig get clusterversion -o yaml
+----
++
+.Example output
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1
+kind: ClusterVersion
+metadata:
+  creationTimestamp: 2019-02-27T22:24:21Z
+  generation: 1
+  name: version
+  resourceVersion: "19927"
+  selfLink: /apis/config.openshift.io/v1/clusterversions/version
+  uid: 6e0f4cf8-3ade-11e9-9034-0a923b47ded4
+spec:
+  channel: stable-4.1
+  clusterID: 5ec312f9-f729-429d-a454-61d4906896ca
+status:
+  availableUpdates: null
+  conditions:
+  - lastTransitionTime: 2019-02-27T22:50:30Z
+    message: Done applying 4.1.1
+    status: "True"
+    type: Available
+  - lastTransitionTime: 2019-02-27T22:50:30Z
+    status: "False"
+    type: Failing
+  - lastTransitionTime: 2019-02-27T22:50:30Z
+    message: Cluster version is 4.1.1
+    status: "False"
+    type: Progressing
+  - lastTransitionTime: 2019-02-27T22:24:31Z
+    message: 'Unable to retrieve available updates: unknown version 4.1.1
+    reason: RemoteFailed
+    status: "False"
+    type: RetrievedUpdates
+  desired:
+    image: registry.svc.ci.openshift.org/openshift/origin-release@sha256:91e6f754975963e7db1a9958075eb609ad226968623939d262d1cf45e9dbc39a
+    version: 4.1.1
+  history:
+  - completionTime: 2019-02-27T22:50:30Z
+    image: registry.svc.ci.openshift.org/openshift/origin-release@sha256:91e6f754975963e7db1a9958075eb609ad226968623939d262d1cf45e9dbc39a
+    startedTime: 2019-02-27T22:24:31Z
+    state: Completed
+    version: 4.1.1
+  observedGeneration: 1
+  versionHash: Wa7as_ik1qE=
+----
+
+. View the conditions by running the following command:
++
+[source,terminal]
+----
+$ oc --kubeconfig=${INSTALL_DIR}/auth/kubeconfig get clusterversion version \
+     -o=jsonpath='{range .status.conditions[*]}{.type}{" "}{.status}{" "}{.message}{"\n"}{end}'
+----
++ 
+Some of most important conditions include `Failing`, `Available` and `Progressing`.
++
+.Example output
+[source,terminal]
+----
+Available True Done applying 4.1.1
+Failing False
+Progressing False Cluster version is 4.0.0-0.alpha-2019-02-26-194020
+RetrievedUpdates False Unable to retrieve available updates: unknown version 4.1.1
+----
+
+. Inspect the `ClusterOperator` object by running the following command:
++
+[source,terminal]
+----
+$ oc --kubeconfig=${INSTALL_DIR}/auth/kubeconfig get clusteroperator
+----
++
+The command returns the status of the cluster Operators.
++
+.Example output
+[source,terminal]
+----
+NAME                                  VERSION   AVAILABLE   PROGRESSING   FAILING   SINCE
+cluster-autoscaler                              True        False         False     17m
```

Comment:
There is also a cluster `baremetal` operator which comes handy in case of baremetal issues (it will indicate that some bare metal components cannot be deployed)